libavfilter/dnn/queue.h

```diff
@@ -22,20 +22,95 @@
 #ifndef AVFILTER_DNN_QUEUE_H
 #define AVFILTER_DNN_QUEUE_H
 
+/**
+ * Linear double-ended data structure
+ */
 typedef struct Queue Queue;
 
+/**
+ * @brief Creates a Queue instance.
+ * It initializes the length of the Queue as 0.
+ *
+ * @return Pointer to the Queue
+ * @retval NULL if allocation fails
+ */
 Queue *ff_queue_create(void);
+
+/**
+ * @brief Destroys the Queue instance.
+ * It also frees all elements of the Queue.
+ */
 void ff_queue_destroy(Queue *q);
 
+/**
+ * @brief Returns the length of the Queue
+ */
 size_t ff_queue_size(Queue *q);
 
+/**
+ * @brief Returns a pointer to the data
+ * at the head of the queue.
+ *
+ * @retval NULL if null pointer was passed
+ * or length of queue is empty
```

Comment:
or queue is empty,  or length of queue is 0

Reply:
Okay